src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -226,6 +257,8 @@ private void initializeHttpContext() {
   public void close() {
     try {
       httpClient.close();
+      if (httpPoolingClient != null)
```

Comment:
should this be splitted into several try/catch blocks because now the `httpPoolingClient` might not be touched if `httpClient.close()` throws an exception.